Written lesson: Account Abstraction 2 code overview (#137)
Cyfrin/Updraft: courses/advanced-foundry/6-account-abstraction/2-code-overview/+page.md

# Account Abstraction Lesson 2: Code Overview

> Welcome to lesson 2 of the Account Abstraction course! In this lesson, we will take a look at the codebase for Ethereum and zkSync. They are minimally structured and serve as a great starting point for learning about account abstraction.

Reviewer: Can we modify it to "These are minimal accounts" rather than "They are minimally structured"

> Welcome to lesson 2 of the Account Abstraction course! In this lesson, we will take a look at the codebase for Ethereum and zkSync. They are minimally structured and serve as a great starting point for learning about account abstraction.

Reviewer: ```suggestion
Welcome to lesson 2 of the Account Abstraction course! In this lesson, we will take a look at the codebase for Ethereum and zkSync. They are minimal accounts and serve as a great starting point for learning about account abstraction.
```

## What You Will Learn

> This is just an overview of things to come. You don't need to know much at this point. However, going through the links below can help to give you a glimpse of what is to come in the course. 

Reviewer: Can we modify 
"You don't need to know much at this point" -> "You don't need to understand fully at this point"

> This is just an overview of things to come. You don't need to know much at this point. However, going through the links below can help to give you a glimpse of what is to come in the course. 

Reviewer: ```suggestion
This is just an overview of things to come. You don't need to understand fully at this point. However, going through the links below can help to give you a glimpse of what is to come in the course. 
```

We will be building two minimal account abstraction accounts. One for Ethereum and another for zkSync.

You can check them out here: [Minimal Accounts](https://github.com/Cyfrin/minimal-account-abstraction/tree/main/src)

Or, you can head straight to the main sources below.

- [README file](https://github.com/Cyfrin/minimal-account-abstraction/blob/main/README.md)
- [Ethereum - `MinimalAccount.sol`](https://github.com/Cyfrin/minimal-account-abstraction/blob/main/src/ethereum/MinimalAccount.sol)
- [zkSync – `ZkMinimalAccount.sol`](https://github.com/Cyfrin/minimal-account-abstraction/blob/main/src/zksync/ZkMinimalAccount.sol)

These two ecosystems are vastly different. The upcoming lessons will allow us to become more familiar with both of them. For now, we can see some examples from the README file. Click on ZkMinimalAccount.sol to see the example deployments for zkSync and MinimalAccount.sol for Ethereum.

![Example Deployments](/2-code-overview/example-deployments.png)

> When you are ready, complete this lesson by watching the video lecture. Then, we will begin our journey with Ethereum minimal account abstraction.  

Reviewer: Since the video is not compulsory, the written content is aiming to both compliment and also provide the same knowledge as in the lessons, lets make the following modification:
_"When you are ready, complete this lesson by watching the video lecture. Then, we will begin our journey with Ethereum minimal account abstraction."_ to be -> 
**_"When you are ready, let's begin our journey with Ethereum minimal account abstraction."_**

Author: @ciaranightingale Thank you for the modification suggestions. They all look good!

> When you are ready, complete this lesson by watching the video lecture. Then, we will begin our journey with Ethereum minimal account abstraction.  

Reviewer: ```suggestion
When you are ready, let's begin our journey with Ethereum minimal account abstraction.  
```
